src/library_wasmfs.js

```diff
@@ -212,8 +212,60 @@ mergeInto(LibraryManager.library, {
         __wasmfs_readdir_finish(state);
         return entries;
       });
+    },
+    optionsToFlags:(opts) => {
+      let flags = 0;
+      const MountFlags = {
+        RDONLY: 0x1,
+        WRONLY: 0x2,
+        RDWR: 0x4,
+        CREATE: 0x8,
+        EXCL: 0x10,
+        TRUNC: 0x20,
+        APPEND: 0x40,
+        IRUGO: cDefine(S_IRUSR) | cDefine(S_IRGRP) | cDefine(S_IROTH),
```

Comment:
`cDefine` isn't a real JS function, but rather a macro that Emscripten's JS library build system expands at link time, so it needs to be marked with curly braces. Also, its argument needs to be a string:

`{{{ cDefine('S_IRUSR') }}}`